### ISSUE TITLE
test(ci): Start up Windows Headless Client with fake Portal messages

### ISSUE THREAD

**Issue by ReactorScram** [CloudFire-LLC/cloudfire#6552]
Would have caught #6551 in CI before #6506 merged

The bug would be triggered if we monitor DNS changes and control DNS at the same time.

**Comment by thomaseizinger** [comment]
Can we have unit tests for these modules instead of attempting to fake portal messages? I am not sure that is a viable path to go down. I am worried that it will cause a lot of glue-code dupllication but I am open to it.

**Comment by ReactorScram** [comment]
Yeah, if we can start DNS control and maybe a tunnel device without connlib then that would be enough
